src/Core/Analysis/IAnalysisStatusNotifier.cs

```diff
@@ -24,10 +24,10 @@ namespace SonarLint.VisualStudio.Core.Analysis
 {
     public interface IAnalysisStatusNotifier
```

Comment:
This is the core change. Previously, there was one singleton class of this notifier. So if I wanted to add which analyzer called it, I would've had to add `string analyzerName` in every method. I think it's ugly.
Instead, I opted to make the implementation of this stateful, and to introduce a factory `IAnalysisStatusNotifierFactory.Create(string filePath)`. Right now it only accepts the filePath, but in the next PR it will receive the analyzer name. WDYT?